File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
let obj = {};
let count = 0;

for (const num of nums) {
  obj[num] = (obj[num] || 0) + 1;
}

for (const num in obj) {
  const occurrences = obj[num];
  count += (occurrences * (occurrences - 1)) / 2;
}
    return count
};

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = 0
        freq_count = {}
        for num in nums:
            if num in freq_count:
                count += freq_count[num]
                freq_count[num] += 1
            else:
                freq_count[num] = 1
        return count
            